From ebe911a7b1aa1e42fa1d147817c58d05565c820a Mon Sep 17 00:00:00 2001
From: JohnnyonFlame <johnnyonflame@hotmail.com>
Date: Wed, 12 Jun 2024 23:23:42 -0300
Subject: [PATCH] Workaround to fix performance regression when the blitter is
 enabled. Unfortunately MSAA on Pixmap surfaces cause major performance
 regressions even on low sample sizes and very small framebuffers. Disable
 MSAA when the blitter is used to workaround this issue.

---
 src/video/mali-fbdev/SDL_malivideo.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/src/video/mali-fbdev/SDL_malivideo.c b/src/video/mali-fbdev/SDL_malivideo.c
index 8c40a1a4f..35d82cf98 100644
--- a/src/video/mali-fbdev/SDL_malivideo.c
+++ b/src/video/mali-fbdev/SDL_malivideo.c
@@ -252,6 +252,8 @@ MALI_EGL_InitPixmapSurfaces(_THIS, SDL_Window *window)
     width = window->w;
     height = window->h;
 
+    _this->gl_config.multisamplebuffers = 0;
+    _this->gl_config.multisamplesamples = 0;
     _this->egl_data->egl_surfacetype = EGL_PIXMAP_BIT;
     if (SDL_EGL_ChooseConfig(_this) != 0) {
         SDL_SetError("mali-fbdev: Unable to find a suitable EGL config");